src/univariate/continuous/rician.jl

```diff
@@ -0,0 +1,175 @@
+"""
+    Rician(ν, σ)
+
+The *Rician distribution* with parameters `ν` and `σ` has probability density function:
+
+```math
+f(x; \\nu, \\sigma) = \\frac{x}{\\sigma^2} \\exp\\left( \\frac{-(x^2 + \\nu^2)}{2\\sigma^2} \\right) I_0\\left( \\frac{x\\nu}{\\sigma^2} \\right).
+```
+
+If shape and scale parameters `K` and `Ω` are given instead, `ν` and `σ` may be computed from them:
+
+```math
+\\sigma = \\sqrt{\\frac{\\Omega}{2(K + 1)}}, \\quad \\nu = \\sigma\\sqrt{2K}
+```
+
+```julia
+Rician()         # Rician distribution with parameters ν=0 and σ=1
+Rician(ν, σ)     # Rician distribution with parameters ν and σ
+
+params(d)        # Get the parameters, i.e. (ν, σ)
+shape(d)         # Get the shape parameter K = ν²/2σ²
+scale(d)         # Get the scale parameter Ω = ν² + 2σ²
+```
+
+External links:
+
+* [Rician distribution on Wikipedia](https://en.wikipedia.org/wiki/Rice_distribution)
+
+"""
+struct Rician{T<:Real} <: ContinuousUnivariateDistribution
+    ν::T
+    σ::T
+    Rician{T}(ν, σ) where {T} = new{T}(ν, σ)
+end
+
+function Rician(ν::T, σ::T; check_args=true) where {T<:Real}
+    check_args && @check_args(Rician, ν ≥ zero(ν) && σ ≥ zero(σ))
+    return Rician{T}(ν, σ)
+end
+
+Rician() = Rician(0.0, 1.0)
+Rician(ν::Real, σ::Real) = Rician(promote(ν, σ)...)
+Rician(ν::Integer, σ::Integer) = Rician(float(ν), float(σ))
+
+@distr_support Rician 0.0 Inf
+
+#### Conversions
+
+function convert(::Type{Rician{T}}, ν::Real, σ::Real) where T<:Real
+    Rician(T(ν), T(σ))
+end
+
+function convert(::Type{Rician{T}}, d::Rician{S}) where {T <: Real, S <: Real}
+    Rician(T(d.ν), T(d.σ); check_args=false)
+end
+
+#### Parameters
+
+shape(d::Rician) = d.ν^2 / (2 * d.σ^2)
+scale(d::Rician) = d.ν^2 + 2 * d.σ^2
+
+params(d::Rician) = (d.ν, d.σ)
+partype(d::Rician{T}) where {T<:Real} = T
+
+#### Statistics
+
+# helper
+_Lhalf(x) = exp(x/2) * ((1-x) * besseli(zero(x), -x/2) - x * besseli(oneunit(x), -x/2))
+
+mean(d::Rician{T}) where T = d.σ * √T(π/2) * _Lhalf(-d.ν^2/(2 * d.σ^2))
+var(d::Rician) = 2 * d.σ^2 + d.ν^2 - (π * d.σ^2 / 2) * _Lhalf(-d.ν^2/(2 * d.σ^2))^2
+
+function mode(d::Rician)
+    m = mean(d)
+    _minimize_gss(x -> -pdf(d, x), zero(m), m)
+end
+
+# helper: 1D minimization using Golden-section search
+function _minimize_gss(f, a, b; tol=1e-12)
+    ϕ = (√5 + 1) / 2
+    c = b - (b - a) / ϕ
+    d = a + (b - a) / ϕ
+    while abs(b - a) > tol
+        if f(c) < f(d)
+            b = d
+        else
+            a = c
+        end
+        c = b - (b - a) / ϕ
+        d = a + (b - a) / ϕ
+    end
+    (b + a) / 2
+end
+
+#### PDF/CDF/quantile delegated to NoncentralChisq
+
+# NOTE:
+# 1.0ν and 1.0x used to support x::Float32, since the underlying NoncentralChisq delegates
+# to StatsFuns(rmath.jl) which doesn't have support for Float32
```

Comment:
I initially tried to keep everything to the precision it was, but realized that many of the other distributions in this package propagate to `Float64`, and that `NoncentralChisq` doesn't even work with `Float32`. So while I agree with you that this should be fixed when `StatsFuns` in fixed, having a broken implementation in this PR seems extreme. My preference was to come back and change the implementation once `StatsFuns` is fixed, but you seem to prefer otherwise. I'll defer to your wisdom on this.

Reply:
Fixed in f8a7847 and other accepted changes.

Reply:
I opened a PR to StatsFuns that fixes the `Float32` issue: https://github.com/JuliaStats/StatsFuns.jl/pull/124